strings/base_meta.h

```diff
@@ -117,23 +117,40 @@ namespace winrt::impl
         static constexpr auto data{ category_signature<category_t<T>, T>::data };
     };
 
-#if defined(__clang__)
     template <typename T>
-    struct classic_com_guid
+    struct classic_com_guid_error
     {
-#if __has_declspec_attribute(uuid) && defined(WINRT_IMPL_IUNKNOWN_DEFINED)
-        static constexpr guid value{ __uuidof(T) };
+#ifdef __clang__
+#if !__has_declspec_attribute(uuid)
+        static_assert(std::is_void_v<T> /* dependent_false */, "To use classic COM interfaces, you must compile with -fms-extensions.");
+#endif 
+#endif
+
+#if defined(_MSC_VER) || defined(__clang__)
+#ifndef WINRT_IMPL_IUNKNOWN_DEFINED
+        static_assert(std::is_void_v<T> /* dependent_false */, "To use classic COM interfaces, you must include <unknwn.h> before including C++/WinRT headers.");
+#endif
 #else
-        static_assert(std::is_void_v<T> /* dependent_false */, "To use classic COM interfaces, you must compile with -fms-extensions and include <unknwn.h> before including C++/WinRT headers.");
+        static_assert(std::is_void_v<T> /* dependent_false */, "Classic COM interfaces are not supported with this compiler.");
 #endif
     };
 
     template <typename T>
-    inline constexpr guid guid_v = classic_com_guid<T>::value;
+#ifdef WINRT_IMPL_IUNKNOWN_DEFINED
+#ifdef __clang__
+#if __has_declspec_attribute(uuid)
+    inline constexpr guid guid_v{ __uuidof(T) };
 #else
-    template <typename T>
+    inline constexpr guid guid_v = classic_com_guid_error<T>::value;
+#endif // __has_declspec_attribute(uuid)
+#elif defined(_MSC_VER)
     inline constexpr guid guid_v{ __uuidof(T) };
```

Comment:
MSVC is forgiving and allows you include `<unknwn.h>` later, as long as you include it before using `guid_v`. So if you're consuming COM interfaces but not producing them, then it's okay to include `<unknwn.h>` afterwards. However, I'm okay with calling that an error anyway.

Reply:
Ah, I was under the impression from https://github.com/microsoft/cppwinrt/pull/1193#issuecomment-1262804938 that including `<unknwn.h>` after `<winrt/base.h>` wouldn't work, and the user would still get the error when `guid_v` gets expanded.

I can fix this, but I'm also okay with making this an error, if it doesn't break too many people. Does an OS build pass with this PR applied as-is?

Reply:
The OS doesn't take every build of cppwinrt as it takes a fair amount of time to validate. If there's an issue with OS ingestion, we can always tweak it as needed.

Reply:
Brought back the ability for MSVC to use guid_v when unknwn.h is included after winrt/base.h. This was valid before #1180, so we avoid needlessly breaking existing code.